### PR TITLE
Expose overlay on the ready event detail

### LINKED ISSUE #168: Expose the overlay instance on the `oOverlay.ready` event
It'd be useful for the overlay ready event to expose the overlay instance. Should be a relatively small change to add it as `event.detail.overlay`.

### PR DESCRIPTION
**Pull request by chee**
I don't know what specific use-case this is for, but:
let's either merge this or decide not to do it and close #168

### CODE REVIEW

**Comment by origamiserviceuser** [comment]
o-overlay bundle size difference from 3.0.10 to 9da059481ad9d358f79e8fda2128dee03fecffb8
No significant bundle size differences found.

**Review by notlee — COMMENTED** [review]
Let's do this 👍 could you document with JSDoc and in the README? Some components have an ["Events" Section](https://github.com/Financial-Times/o-table#events). o-table exposes the table associated with its events with an `instance` property. Think we should standardise that property in the spec?

File: src/js/overlay.js [hunk 1]
```diff
@@ -430,7 +430,9 @@ class Overlay {
 				}
 				this.visible = true;
 				this.wrapper.focus();
-				this.broadcast('ready');
+				this.broadcast('ready', 'oOverlay', {
+					overlay: this
```
Comment (chee) [inline src/js/overlay.js:434]:
```suggestion
					instance: this
```

**Comment by origamiserviceuser** [comment]
o-overlay bundle size difference from 3.0.10 to b0ed77d8547128def825fab3f79ddbef59109e35
No significant bundle size differences found.

**Comment by origamiserviceuser** [comment]
o-overlay bundle size difference from 3.0.10 to 199124a0efbe26f40f36563f46716b2c6f75bb85
No significant bundle size differences found.

File: src/js/overlay.js [hunk 2]
```diff
@@ -430,7 +430,9 @@ class Overlay {
 				}
 				this.visible = true;
 				this.wrapper.focus();
-				this.broadcast('ready');
+				this.broadcast('ready', 'oOverlay', {
```
Comment (chee) [inline src/js/overlay.js:433]:
```suggestion
				/**
				 * Ready event
				 *
				 * @event oOverlay#ready
				 * @type {object}
				 * @property {oOverlay} instance - the firing instance
				 */
				this.broadcast('ready', 'oOverlay', {
```

@notlee like this?

**Review by JakeChampion — APPROVED** [review]

File: src/js/overlay.js (hunk 2 above)
Comment (chee) [reply src/js/overlay.js:433]: and a @fires on the show method?

```
	/**
	 * Show the overlay
	 *
	 * @fires oOverlay#ready
	 */
     show()
```

Comment (notlee) [reply src/js/overlay.js:433]: 👍

**Review by notlee — APPROVED** [review]

**Comment by origamiserviceuser** [comment]
o-overlay bundle size difference from 3.0.10 to 47a01f3f03dfc3e9e6b323d48c750009a7df197c
No significant bundle size differences found.

**Comment by origamiserviceuser** [comment]
o-overlay bundle size difference from 3.0.10 to 78ae88559a34395bbcfc260f3961f7f348cabe93
No significant bundle size differences found.

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v3.1.0 :tada:

**Pull request merged**